### DISCUSSION TITLE
react-native-logs new features

### DISCUSSION THREAD

**Discussion by alessandro-bottamedi** [mowispace/react-native-logs#106] [Ideas]
## 👋 Welcome!

I’m opening this discussion to talk about the future versions of react-native-logs, while always maintaining the focus of the package, which is **simplicity**, **modularity**, and **platform independence**. What features do you think would be useful to add and/or improve?

**Comment by g-otn** [comment]
In regard to modularity, what if the preset transports were not reexported from the module entry?

So you'd had to import it using something like this:
```ts
import sentryTransport from 'react-native-logs/transport/sentryTransport'
```

This way, as more transports are implemented, the bundle only includes the transports used. (with the default one always being included, of course)

It's a minor performance improvement which introduces a breaking change. Maybe with new bundle optimizations in Metro, React Native and Expo, this won't be necessary. However it's how I've seen preset but optional features being exposed in some libraries.

**Comment by g-otn** [comment]
A feature that I think would be nice to have the option to enable logs at specific logging levels in the console transport.

For example, the stack traces are useful in error and warn levels during development.

I don't think `console.trace()` works in React Native (in Hermes for ios/android at least), so we'd have to call `console.warn/error`.

I can see two ways this could be implemented:
- Have a transport option which maps levels to which console function it should use. (debug, log, info, error) Therefore allowing use of 'warn' and 'error' which produce stack traces.
- Pass logging level information when calling `consoleFunc` so this can be implemented outside the transport.

**Reply by g-otn** [reply]
For now I'm using a simple patch which returns the `level` when calling `consoleFunc`, and outside I'm doing something like:
```ts
    consoleFunc: (msg: string, level?: string) => {
      switch (level) {
        case 'error':
          console.error(msg);
          break;
        case 'warn':
          console.warn(msg);
          break;
        case 'info':
          console.info(msg);
          break;
        case 'debug':
          console.debug(msg);
          break;
        default:
          console.log(msg);
      }
    },
  },
```
![image](https://github.com/user-attachments/assets/f71ffa27-6680-44ae-bc78-afd01e182930)

Especially with metro server logs going away in the future, I think this is a neat feature for developing while using `consoleTransport`.

**Reply by g-otn** [reply]
I just noticed this is already implemented with the `mapConsoleTransport` :sweat_smile:, however it doesn't support colors / extensionColors. So I've created an issue: https://github.com/mowispace/react-native-logs/issues/114
